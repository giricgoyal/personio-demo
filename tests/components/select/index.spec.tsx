import { fireEvent, getByTestId } from '@testing-library/react'
import React from 'react'
import Select from 'src/components/select'
import { render } from 'tests/test-utils'

describe('components/select', () => {
    let container, getByTestId, onChangeHandler

    const options = [
        {
            label: 'option 1',
            value: 'option1',
        },
        {
            label: 'Option 2',
            value: 'option2',
        },
    ]
    beforeEach(() => {
        onChangeHandler = jest.fn()
        ;({ container, getByTestId } = render(
            <Select selected="option2" options={options} onChange={onChangeHandler} label="label" />,
        ))
    })

    test('should render and match snapshot', () => {
        expect(container).toMatchSnapshot()
    })

    describe('When select value is changed', () => {
        beforeEach(() => {
            fireEvent.change(getByTestId('select-input'), {
                target: {
                    value: 'option1',
                },
            })
        })

        test('Should trigger onChangeHandler', () => {
            expect(onChangeHandler).toHaveBeenCalledWith('option1')
        })
    })
})
