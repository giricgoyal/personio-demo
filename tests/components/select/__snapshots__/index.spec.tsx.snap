// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/select should render and match snapshot 1`] = `
<div>
  <label
    class="select"
  >
    <span
      class="select__label"
    >
      label
    </span>
    <select
      class="select__element"
      data-testid="select-input"
    >
      <option
        selected=""
        value="option2"
      >
        Option 2
      </option>
      <option
        value="option1"
      >
        option 1
      </option>
    </select>
  </label>
</div>
`;
